Examples/Run/Geant4/GeantinoRecordingBase.hpp

```diff
@@ -1,68 +1,54 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2017-2020 CERN for the benefit of the Acts project
+// Copyright (C) 2020 CERN for the benefit of the Acts project
 //
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-#include "ACTFW/DD4hepDetector/DD4hepDetectorOptions.hpp"
-#include "ACTFW/DD4hepDetector/DD4hepGeometryService.hpp"
+#pragma once
+
+#include <boost/program_options.hpp>
+
 #include "ACTFW/Framework/RandomNumbers.hpp"
 #include "ACTFW/Framework/Sequencer.hpp"
 #include "ACTFW/Io/Root/RootMaterialTrackWriter.hpp"
 #include "ACTFW/Io/Root/RootSimHitWriter.hpp"
 #include "ACTFW/Options/CommonOptions.hpp"
 #include "ACTFW/Utilities/Paths.hpp"
+#include "ActsExamples/Geant4/Geant4Options.hpp"
 #include "ActsExamples/Geant4/GeantinoRecording.hpp"
-#include "ActsExamples/Geant4DD4hep/DD4hepDetectorConstruction.hpp"
-
-#include <boost/program_options.hpp>
+#include "G4VUserDetectorConstruction.hh"
 
 using namespace ActsExamples;
 using namespace FW;
 
-int main(int argc, char* argv[]) {
-  // setup and parse options
-  auto desc = Options::makeDefaultOptions();
-  Options::addSequencerOptions(desc);
-  Options::addOutputOptions(desc);
-  Options::addDD4hepOptions(desc);
-  auto vm = Options::parse(desc, argc, argv);
-  if (vm.empty()) {
-    return EXIT_FAILURE;
-  }
-
+/// @brief method to process a geometry
+/// @param detector The detector descriptor instance
+int runSimulation(const boost::program_options::variables_map& vm,
```

Comment:
Maybe rename the function to `runGeantinoRecording` to avoid any confusion.

Reply:
good call.